lighthouse-cli/bin.js

```diff
@@ -122,6 +122,21 @@ async function begin() {
     cliFlags.extraHeaders = JSON.parse(extraHeadersStr);
   }
 
+  if (cliFlags.extraCookies) {
+    // TODO: LH.Flags.extraCookies is actually a string at this point, but needs to be
+    // copied over to LH.Settings.extraCookies, which is LH.Crdp.Network.Cookies. Force
+    // the conversion here, but long term either the CLI flag or the setting should have
+    // a different name.
+    // @ts-ignore
+    let extraCookiesStr = /** @type {string} */ (cliFlags.extraCookies);
+    // If not a JSON array, assume it's a path to a JSON file.
+    if (extraCookiesStr.substr(0, 1) !== '[') {
+      extraCookiesStr = fs.readFileSync(extraCookiesStr, 'utf-8');
+    }
+
+    cliFlags.extraCookies = JSON.parse(extraCookiesStr);
+  }
```

Comment:
```suggestion

    if (!Array.isArray(cliFlags.extraCookies)) {
      throw new Error('Not a valid JSON array');
    }
    cliFlags.extraCookies.forEach(key => {
      const cookie = cliFlags.extraCookies[key];
      if (! cookie.url || ! cookie.domain) {
        // Default it to the domain value
        cookie.url = url;
      }
    });

  }
```

Reply:
I can suggest this extra check and defaulting for the URL to avoid no-op

Reply:
this is a good idea @RynatSibahatau! I think we would want this in `lighthouse-core/` though so node users get this defaulting benefit too

Reply:
@patrickhulce  Could you take a look 074eee1?